src/legacy/core_plugins/kibana/public/management/sections/settings/advanced_settings.tsx

```diff
@@ -18,17 +18,18 @@
  */
 
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-
 import { Comparators, EuiFlexGroup, EuiFlexItem, EuiSpacer, Query } from '@elastic/eui';
 
 import { CallOuts } from './components/call_outs';
 import { Search } from './components/search';
 import { Form } from './components/form';
 import { AdvancedSettingsVoiceAnnouncement } from './components/advanced_settings_voice_announcement';
+import { IUiSettingsClient } from '../../../../../../../core/public/';
 
 import { getAriaName, toEditableConfig, DEFAULT_CATEGORY } from './lib';
 
+import { FieldSetting, IQuery } from './types';
```

Comment:
`IQuery` should be prefixed with `AdvancedSettings`.
It's confusing in the context of the `SearchBar` `Query` interface.

Reply:
This is providing types for `SearchBar` and `Query`. I added a comment. I'm hesitant to add to an existing type def file since we have more than one in the kibana repo. Curious what you think is the best solution.